### PR TITLE
metrics series 2 : add DefaultMetricsCollector

### PR DESCRIPTION
**Pull request by Henrik-Yao**
## What is the purpose of the change

Add DefaultMetricsCollector

Related pr : https://github.com/apache/dubbo/pull/10331

Related issues : https://github.com/apache/dubbo/issues/9886

Related Documentation : https://qnac8r4ct8.feishu.cn/docx/doxcncmY1gYiRGX9ccX264W6JJd

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — APPROVED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/common/metrics/collector/DefaultMetricsCollector.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    private final Map<MethodMetric, AtomicLong> rtCount = new ConcurrentHashMap<>();
+
+    public DefaultMetricsCollector(ApplicationModel applicationModel) {
+        this.applicationModel = applicationModel;
+        this.applicationName = applicationModel.getApplicationName();
+    }
+
+    public void setCollectEnabled(Boolean collectEnabled) {
+        this.collectEnabled = collectEnabled;
+    }
```
Comment (CrazyHZM) [inline dubbo-common/src/main/java/org/apache/dubbo/common/metrics/collector/DefaultMetricsCollector.java:77]: Will dynamic start/stop be supported here, if so, `collectEnabled` should be an atomic operation, if not, this method should be removed and set in the constructor

File: dubbo-common/src/main/java/org/apache/dubbo/common/metrics/event/MetricsEvent.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * Metric object. (eg. {@link org.apache.dubbo.common.metrics.model.MethodMetric})
+     */
+    protected transient Object source;
+
+    public MetricsEvent(Object source) {
+        if (source == null)
+            throw new IllegalArgumentException("null source");
+
```
Comment (CrazyHZM) [inline dubbo-common/src/main/java/org/apache/dubbo/common/metrics/event/MetricsEvent.java:33]: Note the code style, add `{}`

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/MetricsConstants.java [hunk 3]
```diff
@@ -20,9 +20,87 @@ public interface MetricsConstants {
...
+
+    String[] METRIC_RT_MAX = new String[]{ "rt.max", "Max Response Time" };
+
+    String[] METRIC_RT_TOTAL = new String[]{ "rt.total", "Total Response Time" };
+
+    String[] METRIC_RT_AVG = new String[]{ "rt.avg", "Average Response Time" };
+
+    String[] METRIC_RT_P99 = new String[]{ "rt.p99", "Response Time P99" };
+
+    String[] METRIC_RT_P95 = new String[]{ "rt.p95", "Response Time P95" };
```
[the remark this answers is not in the document]
Comment (Henrik-Yao) [reply dubbo-common/src/main/java/org/apache/dubbo/common/constants/MetricsConstants.java]: Array already split

File: dubbo-common/src/main/java/org/apache/dubbo/common/metrics/collector/DefaultMetricsCollector.java (hunk 1 above)
Comment (Henrik-Yao) [reply dubbo-common/src/main/java/org/apache/dubbo/common/metrics/collector/DefaultMetricsCollector.java]: Atomic operation used.

File: dubbo-common/src/main/java/org/apache/dubbo/common/metrics/event/MetricsEvent.java (hunk 2 above)
Comment (Henrik-Yao) [reply dubbo-common/src/main/java/org/apache/dubbo/common/metrics/event/MetricsEvent.java]: The format has been modified, thanks.

**Pull request merged**